sql-plugin/src/main/scala/org/apache/spark/sql/rapids/AggregateFunctions.scala

```diff
@@ -1461,19 +1461,30 @@ abstract class GpuDecimalAverage(child: Expression, sumDataType: DecimalType)
           sumDataType, nullOnOverflow = true),
     mergeCount.attr)
 
+  // This is here to be bug for bug compatible with Spark. They round in the divide and then cast
+  // the result to the final value. This loses some data in many cases and we need to be able to
+  // match that. This bug appears to have been fixed in Spark 3.4.0.
```

Comment:
Do we need a followup to shim this for Spark >= 3.4?

Reply:
https://github.com/NVIDIA/spark-rapids/issues/6424